app/src/main/java/net/programmierecke/radiodroid2/players/exoplayer/ExoPlayerWrapper.java

```diff
@@ -142,19 +177,16 @@ public void playRemote(@NonNull OkHttpClient httpClient, @NonNull String streamU
         isHls = streamUrl.endsWith(".m3u8");
 
         SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(context.getApplicationContext());
-        final int retryTimeout = prefs.getInt("settings_retry_timeout", 4);
-        final int retryDelay = prefs.getInt("settings_retry_delay", 10);
+        final int retryTimeout = prefs.getInt("settings_retry_timeout", 10);
+        final int retryDelay = prefs.getInt("settings_retry_delay", 100);
 
         DataSource.Factory dataSourceFactory = new RadioDataSourceFactory(httpClient, bandwidthMeter, this, retryTimeout, retryDelay);
         // Produces Extractor instances for parsing the media data.
-        ExtractorsFactory extractorsFactory = new DefaultExtractorsFactory();
-
         if (!isHls) {
-            audioSource = new ExtractorMediaSource.Factory(dataSourceFactory).setExtractorsFactory(extractorsFactory)
-                    .createMediaSource(Uri.parse(streamUrl));
+            audioSource = new ProgressiveMediaSource.Factory(dataSourceFactory).setLoadErrorHandlingPolicy(new CustomLoadErrorHandlingPolicy()).createMediaSource(Uri.parse(streamUrl));
```

Comment:
It would be better to break lines in builder pattern:

```
            audioSource = new ProgressiveMediaSource.Factory(dataSourceFactory)
                    .setLoadErrorHandlingPolicy(new CustomLoadErrorHandlingPolicy())
                    .createMediaSource(Uri.parse(streamUrl));
```

Reply:
OK